### PR TITLE
Simplify structure of :options and support whitelisting of embedded relations fields

### PR DESCRIPTION
**Pull request by JagdeepSingh**
**DO NOT MERGE**

- [x] Fix failing rspecs
- [x] Add rspecs
- [x] Add changes to README and CHANGELOG

### CODE REVIEW

**Comment by johnnyshields** [comment]
@JagdeepSingh can you please rename `tracked_embedded_many?` to `tracked_embeds_many?`

**Comment by johnnyshields** [comment]
@JagdeepSingh I don't think the whitelist of attributes is filtering correctly on embedded relations. Please write specs for this.

**Comment by JagdeepSingh** [comment]
Renaming done.

**Comment by JagdeepSingh** [comment]
@dblock  `rubocop` is giving errors with ruby < 2.x and jruby. One example is, it doesn't recognize `%i` and gives error. But if i replace the `%i` with array of symbols in code, it starts giving errors for ruby >= 2.x ('Please use %i for creating an array of symbols').

I think one solution may be to run `rubocop` only for the latest version of ruby in the builds.

**Comment by dblock** [comment]
I think we can definitely drop Ruby < 2.x support and ignore jruby failures possibly (I'd rather not, but we can if we have to).

**Comment by JagdeepSingh** [comment]
@dblock this is ready to be reviewed.

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,9 @@ rvm:
 
 matrix:
   allow_failures:
+    - rvm: 1.9.3
```
Comment (dblock) [inline .travis.yml]: We should remove this, no need to support anything < ruby 2.0 and maybe even 2.2 I would say.

File: README.md [hunk 2]
```diff
@@ -202,7 +202,32 @@ track.original # {}
...
+**Whitelist the tracked attributes of embedded relations**
+
+If you don't want to track all the attributes of embedded relations in parent audit history, you can whitelist the attributes as below:
+
+```ruby
+class Book
+  include Mongoid::Document
+  ...
+  embeds_many :pages
+  track_history :on => { :pages => [:title, :content] }
```
Comment (dblock) [inline README.md]: Minor: ruby 2.x syntax, `on: { pages: [ :title, :content ] }`.

**Comment by dblock** [comment]
Merging this, excellent work!

File: README.md (hunk 2 above)
Comment (JagdeepSingh) [inline README.md]: @dblock i was thinking of using the new syntax for hash. But to keep consistency with all the other documentation, i used the older one.

Comment (dblock) [inline README.md]: Makes sense.

**Pull request merged**